README.md
# Lottie Animation View for Angular ([React](https://github.com/chenqingspring/react-lottie), [Vue](https://github.com/chenqingspring/vue-lottie))

[![Build Status](https://travis-ci.org/chenqingspring/ng-lottie.svg?branch=master)](https://travis-ci.org/chenqingspring/ng-lottie)
[![npm version](https://badge.fury.io/js/ng-lottie.svg)](http://badge.fury.io/js/ng-lottie)
[![GitHub issues](https://img.shields.io/github/issues/chenqingspring/ng-lottie.svg)](https://github.com/chenqingspring/ng-lottie/issues)
[![GitHub stars](https://img.shields.io/github/stars/chenqingspring/ng-lottie.svg)](https://github.com/chenqingspring/ng-lottie/stargazers)
[![GitHub license](https://img.shields.io/badge/license-MIT-blue.svg)](https://raw.githubusercontent.com/chenqingspring/ng-lottie/master/LICENSE)

## Demo
https://chenqingspring.github.io/ng-lottie/demo/

Renamed from ***lottie-angular2*** after angular4 supported

## Wapper of bodymovin.js

[bodymovin](https://github.com/bodymovin/bodymovin) is [Adobe After Effects](http://www.adobe.com/products/aftereffects.html) plugin for exporting animations as JSON, also it provide bodymovin.js for vender them as svg/canvas/html.

## Why Lottie?

#### Flexible After Effects features
We currently support solids, shape layers, masks, alpha mattes, trim paths, and dash patterns. And we’ll be adding new features on a regular basis.

#### Manipulate your animation any way you like
You can go forward, backward, and most importantly you can program your animation to respond to any interaction.

#### Small file sizes
Bundle vector animations within your app without having to worry about multiple dimensions or large file sizes. Alternatively, you can decouple animation files from your app’s code entirely by loading them from a JSON API.

[Learn more](http://airbnb.design/introducing-lottie/) › http://airbnb.design/lottie/

Looking for lottie files › https://www.lottiefiles.com/

## Installation

Install through npm:
```
npm install --save ng-lottie
```

Then include in your apps module:

```typescript
import { Component, NgModule } from '@angular/core';
import { LottieAnimationViewModule } from 'ng-lottie';

@NgModule({
  imports: [
    LottieAnimationViewModule.forRoot()
  ]
})
export class MyModule {}
```

Finally use in one of your apps components:
```typescript
import {Component} from '@angular/core';

@Component({
    selector: 'lottie-animation-view-demo-app',
    template: ` <lottie-animation-view
                    [options]="lottieConfig"
                    [width]="300"
                    [height]="600"
                    (animCreated)="handleAnimation($event)">
              </lottie-animation-view>
              <div id="player">
                <p>Speed: x{{animationSpeed}}</p>
                <div class="range-container">
                  <input #range type="range" value="1" min="0" max="3" step="0.5"
                    (change)="setSpeed(range.value)">
                </div>
                <button (click)="stop()">stop</button>
                <button (click)="pause()">pause</button>
                <button (click)="play()">play</button>
              </div>`
})

export class DemoComponent {

    public lottieConfig: Object;
    private anim: any;
    private animationSpeed: number = 1;

    constructor() {
        this.lottieConfig = {
            path: 'assets/pinjump.json',
            renderer: 'canvas',
            autoplay: true,
            loop: true
        };
    }

    handleAnimation(anim: any) {
        this.anim = anim;
    }

    stop() {
        this.anim.stop();
    }

    play() {
        this.anim.play();
    }

    pause() {
        this.anim.pause();
    }

    setSpeed(speed: number) {
        this.animationSpeed = speed;
        this.anim.setSpeed(speed);
    }

}

```

## Related Projects

* [Bodymovin](https://github.com/bodymovin/bodymovin) ng-lottie is a wrapper of bodymovin
* [Ionic Lottie](https://github.com/yannbf/ionic-lottie) Thanks for @yannbf demonstrating ng-lottie on ionic3
* [React Lottie](https://github.com/chenqingspring/react-lottie) react implementation
* [Vue Lottie](https://github.com/chenqingspring/vue-lottie) vue implementation
* [React Native Lottie](https://github.com/airbnb/lottie-react-native) react native implementation by airbnb
* [IOS Lottie](https://github.com/airbnb/lottie-ios) ios implementation by airbnb
* [Android Lottie](https://github.com/airbnb/lottie-android) android implementation by airbnb

## Contribution

Your contributions and suggestions are heartily welcome.

## License

MIT


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

<a name="0.3.2"></a>
## [0.3.2](https://github.com/chenqingspring/ng-lottie/compare/v0.3.1...v0.3.2) (2018-08-31)


### Features

* **options:** allow changing `rendered` option ([c79dabe](https://github.com/chenqingspring/ng-lottie/commit/c79dabe))



<a name="0.3.1"></a>
## [0.3.1](https://github.com/chenqingspring/ng-lottie/compare/v0.3.0...v0.3.1) (2018-01-17)



<a name="0.3.0"></a>
# [0.3.0](https://github.com/chenqingspring/ng-lottie/compare/v0.2.2...v0.3.0) (2018-01-17)



<a name="0.2.2"></a>
## [0.2.2](https://github.com/chenqingspring/ng-lottie/compare/v0.2.1...v0.2.2) (2017-07-07)



<a name="0.2.1"></a>
## [0.2.1](https://github.com/chenqingspring/ng-lottie/compare/v0.2.0...v0.2.1) (2017-05-31)



<a name="0.2.0"></a>
# [0.2.0](https://github.com/chenqingspring/ng-lottie/compare/v0.1.0...v0.2.0) (2017-05-10)


### Bug Fixes

* **angular2:** to angular ([a010342](https://github.com/chenqingspring/ng-lottie/commit/a010342))
* **demo:** remove warning ([e05a2b0](https://github.com/chenqingspring/ng-lottie/commit/e05a2b0))



<a name="0.1.0"></a>
# [0.1.0](https://github.com/chenqingspring/ng-lottie/compare/v0.0.9...v0.1.0) (2017-04-19)



<a name="0.0.9"></a>
## [0.0.9](https://github.com/chenqingspring/ng-lottie/compare/v0.0.8...v0.0.9) (2017-04-19)


### Bug Fixes

* **peerDependencies:** delete angular/core ([cd4fb12](https://github.com/chenqingspring/ng-lottie/commit/cd4fb12))



<a name="0.0.8"></a>
## [0.0.8](https://github.com/chenqingspring/ng-lottie/compare/v0.0.7...v0.0.8) (2017-04-19)



<a name="0.0.7"></a>
## [0.0.7](https://github.com/chenqingspring/ng-lottie/compare/v0.0.6...v0.0.7) (2017-04-19)


### Features

* **angular:** upgrade to ng4 ([99c70b7](https://github.com/chenqingspring/ng-lottie/commit/99c70b7))



<a name="0.0.6"></a>
## [0.0.6](https://github.com/chenqingspring/ng-lottie/compare/v0.0.5...v0.0.6) (2017-03-04)



<a name="0.0.5"></a>
## [0.0.5](https://github.com/chenqingspring/ng-lottie/compare/v0.0.4...v0.0.5) (2017-03-03)



<a name="0.0.4"></a>
## [0.0.4](https://github.com/chenqingspring/ng-lottie/compare/v0.0.1...v0.0.4) (2017-03-03)



<a name="0.0.1"></a>
## 0.0.1 (2017-02-15)


demo/demo.component.ts
import { Component } from '@angular/core';

@Component({
    selector: 'lottie-animation-view-demo-app',
    template: ` <lottie-animation-view
                    [options]="lottieConfig"
                    [width]="300"
                    [height]="600"
                    (animCreated)="handleAnimation($event)">
              </lottie-animation-view>
              <div id="player">
                <p>Speed: x{{animationSpeed}}</p>
                <div class="range-container">
                  <input #range type="range" value="1" min="0" max="3" step="0.5"
                    (change)="setSpeed(range.value)">
                </div>
                <button (click)="stop()">stop</button>
                <button (click)="pause()">pause</button>
                <button (click)="play()">play</button>
              </div>`,
    styles: [`
    #player{
       text-align: center;
    }

    .range-container{
      display: flex;
      justify-content: center;
    }

    .range-container input{
      width:200px;
      margin-bottom: 10px;
    }
  `],
})

export class DemoComponent {

    public lottieConfig: Object;
    private anim: any;
    private animationSpeed: number = 1;

    constructor() {
        this.lottieConfig = {
            path: 'assets/pinjump.json',
            autoplay: true,
            loop: true
        };
    }

    handleAnimation(anim: any) {
        this.anim = anim;
    }

    stop() {
        this.anim.stop();
    }

    play() {
        this.anim.play();
    }

    pause() {
        this.anim.pause();
    }

    setSpeed(speed: number) {
        this.animationSpeed = speed;
        this.anim.setSpeed(speed);
    }

}
